x/auth/vesting/types/vesting_account.go

```diff
@@ -41,56 +41,53 @@ type BaseVestingAccount struct {
 	EndTime          int64     `json:"end_time" yaml:"end_time"`                   // when the coins become unlocked
 }
 
-// NewBaseVestingAccount creates a new BaseVestingAccount object
-func NewBaseVestingAccount(baseAccount *authtypes.BaseAccount, originalVesting sdk.Coins, endTime int64) (*BaseVestingAccount, error) {
-	if (baseAccount.Coins.IsZero() && !originalVesting.IsZero()) || originalVesting.IsAnyGT(baseAccount.Coins) {
-		return &BaseVestingAccount{}, errors.New("vesting amount cannot be greater than total amount")
-	}
+// NewBaseVestingAccount creates a new BaseVestingAccount object. It is the
+// callers responsibility to ensure the base account has sufficient funds with
+// regards to the original vesting amount.
+func NewBaseVestingAccount(baseAccount *authtypes.BaseAccount, originalVesting sdk.Coins, endTime int64) *BaseVestingAccount {
 	return &BaseVestingAccount{
 		BaseAccount:      baseAccount,
 		OriginalVesting:  originalVesting,
 		DelegatedFree:    sdk.NewCoins(),
 		DelegatedVesting: sdk.NewCoins(),
 		EndTime:          endTime,
-	}, nil
+	}
 }
 
-// SpendableCoinsVestingAccount returns all the spendable coins for a vesting account given a
-// set of vesting coins.
+// LockedCoinsFromVesting returns all the coins that are not spendable (i.e. locked)
+// for a vesting account given the current vesting coins and account balance,
+// where the balance only includes tokens of vesting denomination.
 //
-// CONTRACT: The account's coins, delegated vesting coins, vestingCoins must be
-// sorted.
-func (bva BaseVestingAccount) SpendableCoinsVestingAccount(vestingCoins sdk.Coins) sdk.Coins {
-	var spendableCoins sdk.Coins
-	bc := bva.GetCoins()
-
-	for _, coin := range bc {
-		baseAmt := coin.Amount
-		vestingAmt := vestingCoins.AmountOf(coin.Denom)
-		delVestingAmt := bva.DelegatedVesting.AmountOf(coin.Denom)
+// CONTRACT: Delegated vesting coins and vestingCoins must be sorted.
+func (bva BaseVestingAccount) LockedCoinsFromVesting(balance, vestingCoins sdk.Coins) sdk.Coins {
+	var lockedCoins sdk.Coins
+
+	for _, vestingCoin := range vestingCoins {
+		vestingAmt := vestingCoins.AmountOf(vestingCoin.Denom)
+		delVestingAmt := bva.DelegatedVesting.AmountOf(vestingCoin.Denom)
+		baseAmt := balance.AmountOf(vestingCoin.Denom)
 
 		// compute min((BC + DV) - V, BC) per the specification
 		min := sdk.MinInt(baseAmt.Add(delVestingAmt).Sub(vestingAmt), baseAmt)
-		spendableCoin := sdk.NewCoin(coin.Denom, min)
+		lockedCoin := sdk.NewCoin(vestingCoin.Denom, baseAmt.Sub(min))
```

Comment:
mhhh, yeah that looks right actually. This would mean we wouldn't even need to pass in balance.

Reply:
This looks correct to me. It's been awhile since I read the spec though, if we update it here we should make sure to also do it elsewhere where the same (or an equivalent) calculation is performed.

Reply:
Updated. I'll update the spec and ADR accordingly.